src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const collectionObjects = [];
+  let objectStep = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(objectStep, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete objectStep[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in objectStep) {
+          delete objectStep[key];
+        }
```

Comment:
It's not necessary to create loop for it.
case 'clear':
        cloneState = {};

Reply:
It`s true! In the last task, it was impossible. But, if we use clone object it`s work! Thanks!